### ISSUE TITLE
capitalized headers are not parsing correctly for h2 requests

### ISSUE THREAD

**Issue by RomanBelozerov** [tempesta-tech/tempesta#1729]
# Scope
h2 request with capitalized headers is forwarded to backend without parsing this header.

For request from deproxy h2 client (disable request parsing for client):
```
(
    [
          (':authority', 'localhost'),
          (':path', '/'),
          (':scheme', 'https'),
          (':method', 'POST'),
          ('Content-Length', '5'),
    ],
    '12345'
)
```
request to backend:
```
POST / HTTP/1.1
host: localhost
x-forwarded-for: 127.0.0.1
Content-Length: 5
via: 1.1 tempesta_fw (Tempesta FW pre-0.7.0)
Content-Length: 5

12345
```

[Test to reproduce.](https://github.com/tempesta-tech/tempesta-test/blob/ebf95137bf63b78b389b58f1915b4b25d8fdc212/http2_general/test_h2_headers.py#L535). And [h2spec](https://github.com/tempesta-tech/tempesta-test/blob/f6fd79d9800e067c2248fc7ae0982d625b1cb19b/http2_general/test_h2_specs.py#L123) test

**Comment by const-t** [comment]
References: 
Old RFC https://www.rfc-editor.org/rfc/rfc7540#section-8.1.2
New RFC https://www.rfc-editor.org/rfc/rfc9113.html#name-http-fields
We have some difference.

**Comment by krizhanovsky** [comment]
Closed in #1903
